src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+
+import { GlobalOptions } from '../types';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function resolveHome(home?: string) {
+  if (home) {
+    return home;
+  }
+
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  // TODO: Resolve options with precedence (env, command line etc.)
```

Comment:
Yup. I think we can add some `resolveEnv` functions later. I would propose something like this:

```ts
const home = resolveHome(programOptions.home);
// this returns `Record<string, string | undefined>`, parsing the app.yaml file in home
// if file is missing, just returns {} (no error)
const appConfig = loadAppConfig(home);
// resolveEnv takes the various items to look up. Throws an error if all are undefined (second read from env)
const src = resolveEnv(programOptions.src, 'RELAYER_SRC', appConfig.src);
const dest = resolveEnv(programOptions.dest, 'RELAYER_DEST', appConfig.dest);

const options = {src, dest, home};
```

Reply:
I thought of a similar thing, gonna do :) I just didn't touch this yet.